circleci/docker-publish

```diff
@@ -78,6 +78,3 @@ if [ "$ECR_PUSH_ID" != "" ]; then
   echo "ECR access_id available. Pushing to ECR..."
   push_ecr_image
 fi
-
```

Comment:
Why does this get removed in this change?

Reply:
This line was added recently during lambda experiments, in order to make the old `docker-publish` parallel to the new `lambda-publish` script. However, @rgarcia realized it's not possible to use the artifact until after the build is complete, so we're removing the recently added line that creates an artifact.

Reply:
this creates a circle ci artifact, which was part of my first attempt at publishing a slightly-different catapult app, along with the deleted file also in this PR